src/XMakeCommandLine/UnitTests/XMake_Tests.cs

```diff
@@ -633,7 +632,8 @@ public void ConfigurationInvalid()
                     "</Project>";
                 File.WriteAllText(pathToProjectFile, projectString);
 
-                output = RunProcessAndGetOutput(newPathToMSBuildExe, "\"" + pathToProjectFile + "\"", expectSuccess: false);
+                var msbuildParameters = "\"" + pathToProjectFile + "\"";
+                output = runMSBuildWithinCoreRun(newPathToMSBuildExe, msbuildParameters, expectSuccess: false);
```

Comment:
True. I'll make a delegate out of that method that depending on an ifdef, gets instantiated with one function or the other